server/src/main/java/org/apache/iotdb/db/qp/executor/PlanExecutor.java

```diff
@@ -772,8 +779,12 @@ private void checkPathExists(MNode node, String fullPath, MeasurementSchema sche
                   measurementNode.getSchema().getType()));
         }
       }
-      return storageEngine.insertBatch(batchInsertPlan);
-
+      Integer[] results = storageEngine.insertBatch(batchInsertPlan);
+      for (int i = 0; i < measurementList.length; i++) {
+        MNode measurementNode = node.getChild(measurementList[i]);
+        measurementNode.updateCachedLast(batchInsertPlan.composeLastTimeValuePair(i));
```

Comment:
Please check this condition:

insert into two INT32 series s1, s2 with values 1, "b". 
These two values will all be cached in the MNode.

However, when inserting to the memtable, the inserting "b" to s2 will throw RuntimeException.
You'd better check this value type before adding it to cache. (The MNode is classified into three type in mmanager_refactor branch, the measurement node is a LeafNode, you could merge this branch)

Refer to AbstractMemtable.insert()